src/circuit/gadgets/constraints.rs

```diff
@@ -1282,4 +1319,32 @@ mod tests {
 
         assert!(cs.is_satisfied());
     }
+
+    #[test]
+    fn test_enforce_u64() {
+        let mut cs = TestConstraintSystem::<Fr>::new();
+
+        let alloc_num = AllocatedNum::alloc(&mut cs.namespace(|| "num"), || {
+            Ok(Fr::from_str_vartime("42").unwrap())
+        })
+        .unwrap();
```

Comment:
Even if not a full-on proptest, I would love to see this test made at least a little more convincing. For example, can we call it on 0, 1, and the largest U64?

Reply:
thanks, just added it